packages/inferno-hooks-generator/src/inferno-component.ts

```diff
@@ -0,0 +1,128 @@
+import { ReactComponent, GetAccessor } from '@devextreme-generator/react';
+import { ObjectLiteral, isComponentWrapper, lowerizeFirstLetter } from '@devextreme-generator/core';
+
+export class InfernoComponent extends ReactComponent {
+  compileImportStatements(hooks: string[], compats: string[], core: string[]): string[] {
+    const componentImports = this.jQueryRegistered
+      ? ['InfernoWrapperComponent', 'useReRenderEffect']
+      : ['HookComponent'];
+
+    const imports = [];
+    const namedImports = hooks
+      .concat(core)
+      .concat(componentImports);
+    if (namedImports.length) {
+      imports.push(`import {${namedImports.join(',')}} from '@devextreme/runtime/inferno-hooks'`);
+    }
+    imports.push('import { normalizeStyles } from \'@devextreme/runtime/common\'');
+    if (this.props.some((p) => p.isTemplate) && !isComponentWrapper(this.context.imports)) {
+      const runTimeImports = isComponentWrapper(this.context.imports) ? 'getWrapperTemplate' : 'getTemplate';
+      imports.push(`import { ${runTimeImports} } from '@devextreme/runtime/react'`);
+    }
+    if (compats.length > 0) {
+      imports.push(`import {${compats.join(', ')}} from 'inferno'`);
+    }
+    return imports;
+  }
+
+  get jQueryRegistered(): boolean {
+    const jqueryProp = this.decorators[0].getParameter('jQuery') as
+          | ObjectLiteral
+          | undefined;
+    return jqueryProp?.getProperty('register')?.toString() === 'true';
+  }
+
+  get forwardRefApiType(): string {
+    return this.members.find((m) => m.inherited && m.isApiRef)
+      ?.typeDeclaration() || 'any';
+  }
+  // if have apiMethod -> (ref)=>(props)=>{}
+  // if have forwardRef -> forwardref((props, ref)=>{})
+  // if jqueryRegistered -> createRerenderEffect
+
+  compileHooksWrapper(): string {
+    const name = this.name;
+
+    const createRerender = `function createRerenderEffect(ref: ${this.forwardRefApiType}){
+        useReRenderEffect()
+        ${name}Fn(ref);
+      }`;
+
+    if (this.hasApiMethod) {
+      return `let refs = new Map();
```

Comment:
Move inlined memoization ref function into the runtime and use WeakMap instead of Map

Reply:
this will be removed in next pr